stable/restore/templates/_helpers.tpl

```diff
@@ -126,6 +126,76 @@ imagePullSecrets:
 {{- end -}}
 {{- end -}}
 
+{{/*
+Get Pod securityContext
+*/}}
+{{- define "securityContext" -}}
+{{- if eq (include "defaultfalse" .Values.database.securityContext.enabled) "true" }}
+securityContext:
+  runAsUser: {{ default 1000 .Values.database.securityContext.runAsUser }}
+  runAsGroup: 0
+  fsGroup: {{ default 1000 .Values.database.securityContext.fsGroup }}
+  {{- include "sc.fsGroupChangePolicy" . }}
+{{- else if eq (include "defaultfalse" .Values.database.securityContext.runAsNonRootGroup) "true" }}
+securityContext:
+  runAsUser: 1000
+  runAsGroup: 1000
+  fsGroup: {{ default 1000 .Values.database.securityContext.fsGroup }}
+  {{- include "sc.fsGroupChangePolicy" . }}
+{{- else if eq (include "defaultfalse" .Values.database.securityContext.fsGroupOnly) "true" }}
+securityContext:
+  fsGroup: {{ default 1000 .Values.database.securityContext.fsGroup }}
+  {{- include "sc.fsGroupChangePolicy" . }}
+{{- end }}
+{{- end -}}
+
+{{/*
+Get fsGroupChangePolicy if Kubernetes version supports it
+*/}}
+{{- define "sc.fsGroupChangePolicy" -}}
+{{- if semverCompare ">=1.20" .Capabilities.KubeVersion.Version }}
+  fsGroupChangePolicy: OnRootMismatch
```

Comment:
It looks like the default value for this is `Always`. Are we specifying this to overcome the slow `chmod -R` for the archive volume?

Reply:
Yes, this is supposed to avoid the issue in `fsGroup` analogous to what @bkelly-ndb discovered with the init container. The reason why it is conditionalized on version is that it does not exist before Kubernetes 1.18 and is in alpha before 1.20.